airflow/models/taskinstance.py

```diff
@@ -1297,9 +1297,10 @@ def signal_handler(signum, frame):  # pylint: disable=unused-argument
                     registered = task_copy.register_in_sensor_service(self, context)
                 except Exception as e:
                     self.log.warning(
-                        "Failed to register in sensor service.Continue to run task in non smart sensor mode."
+                        "Failed to register in sensor service."
+                        "Continue to run task in non smart sensor mode.",
+                        exc_info=True
                     )
-                    self.log.exception(e, exc_info=True)
```

Comment:
This (and some more other changes) makes the message warning-only (previously there’d be one warning and one error message each) and may affect compatibility.

Reply:
Removed `as` in aa94e4a556776daaa7c65df68abd0ea451e5ec2f. Thanks for your advice!

All `logging` methods, include `warning`, supports `exc_info` keyword argument.
So for those statements, I just added the argument.
```patch
- log.warning(msg)
+ log.warning(msg, exc_info=True)
```

Reply:
er, I may misunderstood.

Do you mean that there is use case to set logging level to hide/skip the warning `msg` but keep the `exception` and its trace?

I didn't tweak logging level before in my need, would like to hear that in more details.

Reply:
Yeah, that’s what I was thinking. If a user (say) sets logging level to `ERROR`, previously they’d be able to see info associated with the exception. Not anymore with this patch.

An obvious fix would be to change this to `log.error(msg, exc_info=True)` instead, but that’d of course also make the warning become an error, which also may be problematic.

Personally I think it’s probably best to keep the exception info a warning (i.e. your current approach) since it doesn’t really make that much sense to me to only show the exception traceback but not the main warning message, but this probably should be a decision made by someone with more knowledge on Airflow’s logging configs.

Reply:
Generally, I agree with you. It's not acceptable to drop critical information.
```py
log.warning('descriptive message')
log.exception('critical message')  # hide or downgrade this is unacceptable
```

But in these specific situations (*total 6 cases in this PR*),
all of them (*`exception`*) didn't provide useful (*than their `warning` partners*) information,
but looks really like to **attach** stack trace. I'd believe they are a bit of misuse.

Reply:
I don’t disagree the current approach is bad. But that does not mean it is appropriate to change it to

```python
log.warning('descriptive message', exc_info=True)
```

either (it *also* does not mean it is inappropriate). The correct fix may be

```python
log.exception('descriptive message')
```

instead.

I don’t know the answer, so we need a more experienced reviewer on this one.

Reply:
This change as a warning makes sense. 👍🏻